src/goto-instrument/contracts/contracts.cpp

```diff
@@ -187,26 +187,7 @@ void code_contractst::check_apply_loop_contracts(
 
```

Comment:
Yes it captures a "set" of all the instructions (not "list"), but also, we need to find the "loop_end" instruction where we have a back edge (so that we can remove it).

As to whether this loop is reliable for finding the "loop_end", I think it is? Do you see a case where this may not work?

Reply:
I was thinking we should maybe search for loop end within the set of instructions returned by natural loops. In the general case don't think that loops always correspond to spans of consecutive instructions.